server/src/main/java/io/crate/analyze/CopyAnalyzer.java

```diff
@@ -62,6 +66,9 @@ AnalyzedCopyFrom analyzeCopyFrom(CopyFrom<Expression> node,
             txnCtx.sessionContext().sessionUser(),
             txnCtx.sessionContext().searchPath());
 
+        List<Reference> targetColumns =
+            new ArrayList<>(InsertAnalyzer.resolveTargetColumns(node.columns(), tableInfo));
```

Comment:
I am not sure I understand the problem. 
Inside `AstBuilder` there is:

```
@Override
    public Node visitCopyFrom(SqlBaseParser.CopyFromContext context) {
        boolean returnSummary = context.SUMMARY() != null;
        return new CopyFrom(
            (Table) visit(context.tableWithPartition()),
            context.ident() == null ? emptyList() : identsToStrings(context.ident()),
            visit(context.path),
            extractGenericProperties(context.withProperties()),
            returnSummary);
    }
```

and inside `resolveTargetColumns`

```
if (targetColumnNames.isEmpty()) {
            return targetTable.columns();
        }
```

So we always have some user defined target columns or an empty list which is resolved to all table columns.
Can you please elaborate more on this?

Reply:
If I understand the code correctly it would disable the implicit column creation.

Consider a table like this:

```sql
create table tbl (
  a int,
  b int
)
```


With a CSV file:

```
a,b,c
1,2,3
```

```
COPY .. FROM ... WITH (headers = true)
````

Should still add the `c` column, but if you define the columns `(a, b)` it should not.
So no columns specified is not the same as all columns specified.

Reply:
This is now unused? And in other places there is now some uncommented code?

Reply:
I removed the resolving completely and added a header check inside `CopyFromPlan`. Sorry for the unused code, i removed it! 😅

Do you think that resolving is needed?